openstudiocore/src/geographic_lib/man/GeoConvert.1
.\" Automatically generated by Pod::Man 2.28 (Pod::Simple 3.29)
.\"
.\" Standard preamble:
.\" ========================================================================
.de Sp \" Vertical space (when we can't use .PP)
.if t .sp .5v
.if n .sp
..
.de Vb \" Begin verbatim text
.ft CW
.nf
.ne \\$1
..
.de Ve \" End verbatim text
.ft R
.fi
..
.\" Set up some character translations and predefined strings.  \*(-- will
.\" give an unbreakable dash, \*(PI will give pi, \*(L" will give a left
.\" double quote, and \*(R" will give a right double quote.  \*(C+ will
.\" give a nicer C++.  Capital omega is used to do unbreakable dashes and
.\" therefore won't be available.  \*(C` and \*(C' expand to `' in nroff,
.\" nothing in troff, for use with C<>.
.tr \(*W-
.ds C+ C\v'-.1v'\h'-1p'\s-2+\h'-1p'+\s0\v'.1v'\h'-1p'
.ie n \{\
.    ds -- \(*W-
.    ds PI pi
.    if (\n(.H=4u)&(1m=24u) .ds -- \(*W\h'-12u'\(*W\h'-12u'-\" diablo 10 pitch
.    if (\n(.H=4u)&(1m=20u) .ds -- \(*W\h'-12u'\(*W\h'-8u'-\"  diablo 12 pitch
.    ds L" ""
.    ds R" ""
.    ds C` ""
.    ds C' ""
'br\}
.el\{\
.    ds -- \|\(em\|
.    ds PI \(*p
.    ds L" ``
.    ds R" ''
.    ds C`
.    ds C'
'br\}
.\"
.\" Escape single quotes in literal strings from groff's Unicode transform.
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.\"
.\" If the F register is turned on, we'll generate index entries on stderr for
.\" titles (.TH), headers (.SH), subsections (.SS), items (.Ip), and index
.\" entries marked with X<> in POD.  Of course, you'll have to process the
.\" output yourself in some meaningful fashion.
.\"
.\" Avoid warning from groff about undefined register 'F'.
.de IX
..
.nr rF 0
.if \n(.g .if rF .nr rF 1
.if (\n(rF:(\n(.g==0)) \{
.    if \nF \{
.        de IX
.        tm Index:\\$1\t\\n%\t"\\$2"
..
.        if !\nF==2 \{
.            nr % 0
.            nr F 2
.        \}
.    \}
.\}
.rr rF
.\"
.\" Accent mark definitions (@(#)ms.acc 1.5 88/02/08 SMI; from UCB 4.2).
.\" Fear.  Run.  Save yourself.  No user-serviceable parts.
.    \" fudge factors for nroff and troff
.if n \{\
.    ds #H 0
.    ds #V .8m
.    ds #F .3m
.    ds #[ \f1
.    ds #] \fP
.\}
.if t \{\
.    ds #H ((1u-(\\\\n(.fu%2u))*.13m)
.    ds #V .6m
.    ds #F 0
.    ds #[ \&
.    ds #] \&
.\}
.    \" simple accents for nroff and troff
.if n \{\
.    ds ' \&
.    ds ` \&
.    ds ^ \&
.    ds , \&
.    ds ~ ~
.    ds /
.\}
.if t \{\
.    ds ' \\k:\h'-(\\n(.wu*8/10-\*(#H)'\'\h"|\\n:u"
.    ds ` \\k:\h'-(\\n(.wu*8/10-\*(#H)'\`\h'|\\n:u'
.    ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'^\h'|\\n:u'
.    ds , \\k:\h'-(\\n(.wu*8/10)',\h'|\\n:u'
.    ds ~ \\k:\h'-(\\n(.wu-\*(#H-.1m)'~\h'|\\n:u'
.    ds / \\k:\h'-(\\n(.wu*8/10-\*(#H)'\z\(sl\h'|\\n:u'
.\}
.    \" troff and (daisy-wheel) nroff accents
.ds : \\k:\h'-(\\n(.wu*8/10-\*(#H+.1m+\*(#F)'\v'-\*(#V'\z.\h'.2m+\*(#F'.\h'|\\n:u'\v'\*(#V'
.ds 8 \h'\*(#H'\(*b\h'-\*(#H'
.ds o \\k:\h'-(\\n(.wu+\w'\(de'u-\*(#H)/2u'\v'-.3n'\*(#[\z\(de\v'.3n'\h'|\\n:u'\*(#]
.ds d- \h'\*(#H'\(pd\h'-\w'~'u'\v'-.25m'\f2\(hy\fP\v'.25m'\h'-\*(#H'
.ds D- D\\k:\h'-\w'D'u'\v'-.11m'\z\(hy\v'.11m'\h'|\\n:u'
.ds th \*(#[\v'.3m'\s+1I\s-1\v'-.3m'\h'-(\w'I'u*2/3)'\s-1o\s+1\*(#]
.ds Th \*(#[\s+2I\s-2\h'-\w'I'u*3/5'\v'-.3m'o\v'.3m'\*(#]
.ds ae a\h'-(\w'a'u*4/10)'e
.ds Ae A\h'-(\w'A'u*4/10)'E
.    \" corrections for vroff
.if v .ds ~ \\k:\h'-(\\n(.wu*9/10-\*(#H)'\s-2\u~\d\s+2\h'|\\n:u'
.if v .ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'\v'-.4m'^\v'.4m'\h'|\\n:u'
.    \" for low resolution devices (crt and lpr)
.if \n(.H>23 .if \n(.V>19 \
\{\
.    ds : e
.    ds 8 ss
.    ds o a
.    ds d- d\h'-1'\(ga
.    ds D- D\h'-1'\(hy
.    ds th \o'bp'
.    ds Th \o'LP'
.    ds ae ae
.    ds Ae AE
.\}
.rm #[ #] #H #V #F C
.\" ========================================================================
.\"
.IX Title "GEOCONVERT 1"
.TH GEOCONVERT 1 "2015-09-30" "GeographicLib 1.45" "GeographicLib Utilities"
.\" For nroff, turn off justification.  Always turn off hyphenation; it makes
.\" way too many mistakes in technical documents.
.if n .ad l
.nh
.SH "NAME"
GeoConvert \-\- convert geographic coordinates
.SH "SYNOPSIS"
.IX Header "SYNOPSIS"
\&\fBGeoConvert\fR [ \fB\-g\fR | \fB\-d\fR | \fB\-:\fR | \fB\-u\fR | \fB\-m\fR | \fB\-c\fR ]
[ \fB\-z\fR \fIzone\fR | \fB\-s\fR | \fB\-t\fR ] [ \fB\-n\fR ]
[ \fB\-w\fR ] [ \fB\-p\fR \fIprec\fR ] [ \fB\-l\fR | \fB\-a\fR ]
[ \fB\-\-comment\-delimiter\fR \fIcommentdelim\fR ]
[ \fB\-\-version\fR | \fB\-h\fR | \fB\-\-help\fR ]
[ \fB\-\-input\-file\fR \fIinfile\fR | \fB\-\-input\-string\fR \fIinstring\fR ]
[ \fB\-\-line\-separator\fR \fIlinesep\fR ]
[ \fB\-\-output\-file\fR \fIoutfile\fR ]
.SH "DESCRIPTION"
.IX Header "DESCRIPTION"
\&\fBGeoConvert\fR reads from standard input interpreting each line as a
geographic coordinate and prints the coordinate in the format specified
by the options on standard output.  The input is interpreted in one of
three different ways depending on how many space or comma delimited
tokens there are on the line.  The options \fB\-g\fR, \fB\-d\fR, \fB\-u\fR, and \fB\-m\fR
govern the format of output.  In all cases, the \s-1WGS84\s0 model of the earth
is used (\fIa\fR = 6378137 m, \fIf\fR = 1/298.257223563).
.IP "\fBgeographic\fR" 4
.IX Item "geographic"
2 tokens (output options \fB\-g\fR, \fB\-d\fR, or \fB\-:\fR) given as \fIlatitude\fR
\&\fIlongitude\fR using decimal degrees or degrees, minutes, and seconds.
Latitude is given first (unless the \fB\-w\fR option is given).  See
\&\*(L"\s-1GEOGRAPHIC COORDINATES\*(R"\s0 for a description of the format.  For
example, the following are all equivalent
.Sp
.Vb 5
\&    33.3 44.4
\&    E44.4 N33.3
\&    33d18\*(AqN 44d24\*(AqE
\&    44d24 33d18N
\&    33:18 +44:24
.Ve
.IP "\fB\s-1UTM/UPS\s0\fR" 4
.IX Item "UTM/UPS"
3 tokens (output option \fB\-u\fR) given as \fIzone\fR+\fIhemisphere\fR \fIeasting\fR
\&\fInorthing\fR or \fIeasting\fR \fInorthing\fR \fIzone\fR+\fIhemisphere\fR, where
\&\fIhemisphere\fR is either \fIn\fR (or \fInorth\fR) or \fIs\fR (or \fIsouth\fR).  The
\&\fIzone\fR is absent for a \s-1UPS\s0 specification.  For example,
.Sp
.Vb 4
\&    38n 444140.54 3684706.36
\&    444140.54 3684706.36 38n
\&    s 2173854.98 2985980.58
\&    2173854.98 2985980.58 s
.Ve
.IP "\fB\s-1MRGS\s0\fR" 4
.IX Item "MRGS"
1 token (output option \fB\-m\fR) is used to specify the center of an \s-1MGRS\s0
grid square.  For example,
.Sp
.Vb 2
\&    38SMB4484
\&    38SMB44140847064
.Ve
.SH "OPTIONS"
.IX Header "OPTIONS"
.IP "\fB\-g\fR" 4
.IX Item "-g"
output latitude and longitude using decimal degrees.  Default output mode.
.IP "\fB\-d\fR" 4
.IX Item "-d"
output latitude and longitude using degrees, minutes, and seconds (\s-1DMS\s0).
.IP "\fB\-:\fR" 4
.IX Item "-:"
like \fB\-d\fR, except use : as a separator instead of the d, ', and "
delimiters.
.IP "\fB\-u\fR" 4
.IX Item "-u"
output \s-1UTM\s0 or \s-1UPS.\s0
.IP "\fB\-m\fR" 4
.IX Item "-m"
output \s-1MGRS.\s0
.IP "\fB\-c\fR" 4
.IX Item "-c"
output meridian convergence and scale for the corresponding \s-1UTM\s0 or
\&\s-1UPS\s0 projection.  Convergence is the bearing of grid north given as
degrees clockwise from true north.
.IP "\fB\-z\fR" 4
.IX Item "-z"
set the zone to \fIzone\fR for output.  Use either 0 < \fIzone\fR <=
60 for a \s-1UTM\s0 zone or \fIzone\fR = 0 for \s-1UPS. \s0 Alternatively use a
\&\fIzone\fR+\fIhemisphere\fR designation, e.g., 38n.  See \*(L"\s-1ZONE\*(R"\s0.
.IP "\fB\-s\fR" 4
.IX Item "-s"
use the standard \s-1UPS\s0 and \s-1UTM\s0 zones.
.IP "\fB\-t\fR" 4
.IX Item "-t"
similar to \fB\-s\fR but forces \s-1UPS\s0 regions to the closest \s-1UTM\s0 zone.
.IP "\fB\-n\fR" 4
.IX Item "-n"
on input, \s-1MGRS\s0 coordinates refer to the south-west corner of the \s-1MGRS\s0
square instead of the center; see \*(L"\s-1MGRS\*(R"\s0.
.IP "\fB\-w\fR" 4
.IX Item "-w"
on input and output, longitude precedes latitude (except that, on input,
this can be overridden by a hemisphere designator, \fIN\fR, \fIS\fR, \fIE\fR,
\&\fIW\fR).
.IP "\fB\-p\fR" 4
.IX Item "-p"
set the output precision to \fIprec\fR (default 0); \fIprec\fR is the
precision relative to 1 m.  See \*(L"\s-1PRECISION\*(R"\s0.
.IP "\fB\-l\fR" 4
.IX Item "-l"
on output, \s-1UTM/UPS\s0 uses the long forms \fInorth\fR and \fIsouth\fR to
designate the hemisphere instead of \fIn\fR or \fIs\fR.
.IP "\fB\-a\fR" 4
.IX Item "-a"
on output, \s-1UTM/UPS\s0 uses the abbreviations \fIn\fR and \fIs\fR to designate the
hemisphere instead of \fInorth\fR or \fIsouth\fR; this is the default
representation.
.IP "\fB\-\-comment\-delimiter\fR" 4
.IX Item "--comment-delimiter"
set the comment delimiter to \fIcommentdelim\fR (e.g., \*(L"#\*(R" or \*(L"//\*(R").  If
set, the input lines will be scanned for this delimiter and, if found,
the delimiter and the rest of the line will be removed prior to
processing and subsequently appended to the output line (separated by a
space).
.IP "\fB\-\-version\fR" 4
.IX Item "--version"
print version and exit.
.IP "\fB\-h\fR" 4
.IX Item "-h"
print usage and exit.
.IP "\fB\-\-help\fR" 4
.IX Item "--help"
print full documentation and exit.
.IP "\fB\-\-input\-file\fR" 4
.IX Item "--input-file"
read input from the file \fIinfile\fR instead of from standard input; a file
name of \*(L"\-\*(R" stands for standard input.
.IP "\fB\-\-input\-string\fR" 4
.IX Item "--input-string"
read input from the string \fIinstring\fR instead of from standard input.
All occurrences of the line separator character (default is a semicolon)
in \fIinstring\fR are converted to newlines before the reading begins.
.IP "\fB\-\-line\-separator\fR" 4
.IX Item "--line-separator"
set the line separator character to \fIlinesep\fR.  By default this is a
semicolon.
.IP "\fB\-\-output\-file\fR" 4
.IX Item "--output-file"
write output to the file \fIoutfile\fR instead of to standard output; a
file name of \*(L"\-\*(R" stands for standard output.
.SH "PRECISION"
.IX Header "PRECISION"
\&\fIprec\fR gives precision of the output with \fIprec\fR = 0 giving 1 m
precision, \fIprec\fR = 3 giving 1 mm precision, etc.  \fIprec\fR is the number
of digits after the decimal point for \s-1UTM/UPS. \s0 The number of digits per
coordinate for \s-1MGRS\s0 is 5 + \fIprec\fR.  For decimal degrees, the number of
digits after the decimal point is 5 + \fIprec\fR.  For \s-1DMS \s0(degree, minute,
seconds) output, the number of digits after the decimal point in the
seconds components is 1 + \fIprec\fR; if this is negative then use minutes
(\fIprec\fR = \-2 or \-3) or degrees (\fIprec\fR <= \-4) as the least significant
component.  Print convergence, resp. scale, with 5 + \fIprec\fR, resp. 7 +
\&\fIprec\fR, digits after the decimal point.  The minimum value of \fIprec\fR is
\&\-5 and the maximum is 9 for \s-1UTM/UPS, 9\s0 for decimal degrees, 10 for \s-1DMS,
6\s0 for \s-1MGRS,\s0 and 8 for convergence and scale.
.SH "GEOGRAPHIC COORDINATES"
.IX Header "GEOGRAPHIC COORDINATES"
The utility accepts geographic coordinates, latitude and longitude, in a
number of common formats.  Latitude precedes longitude, unless the \fB\-w\fR
option is given which switches this convention.  On input, either
coordinate may be given first by appending or prepending \fIN\fR or \fIS\fR to
the latitude and \fIE\fR or \fIW\fR to the longitude.  These hemisphere
designators carry an implied sign, positive for \fIN\fR and \fIE\fR and
negative for \fIS\fR and \fIW\fR.  This sign multiplies any +/\- sign prefixing
the coordinate.  The coordinates may be given as decimal degree or as
degrees, minutes, and seconds.  d, ', and " are used to denote degrees,
minutes, and seconds, with the least significant designator optional.
(See \*(L"\s-1QUOTING\*(R"\s0 for how to quote the characters ' and " when entering
coordinates on the command line.)  Alternatively, : (colon) may be used
to separate the various components.  Only the final component of
coordinate can include a decimal point, and the minutes and seconds
components must be less than 60.
.PP
It is also possible to carry out addition or subtraction operations in
geographic coordinates.  If the coordinate includes interior signs
(i.e., not at the beginning or immediately after an initial hemisphere
designator), then the coordinate is split before such signs; the pieces
are parsed separately and the results summed.  For example the point 15"
east of 39N 70W is
.PP
.Vb 1
\&    39N 70W+0:0:15E
.Ve
.PP
\&\fB\s-1WARNING:\s0\fR \*(L"Exponential\*(R" notation is not recognized for geographic
coordinates; instead 7.0E+1 is parsed as (7.0E) + (+1), yielding the
same result as 8.0E.
.PP
Various unicode characters (encoded with \s-1UTF\-8\s0) may also be used to
denote degrees, minutes, and seconds, e.g., the degree, prime, and
double prime symbols; in addition two single quotes can be used to
represent ".
.PP
The other GeographicLib utilities use the same rules for interpreting
geographic coordinates; in addition, azimuths and arc lengths are
interpreted the same way.
.SH "QUOTING"
.IX Header "QUOTING"
Unfortunately the characters ' and \*(L" have special meanings in many
shells and have to be entered with care.  However note (1) that the
trailing designator is optional and that (2) you can use colons as a
separator character.  Thus 10d20' can be entered as 10d20 or 10:20 and
10d20'30\*(R" can be entered as 10:20:30.
.IP "Unix shells (sh, bash, tsch)" 4
.IX Item "Unix shells (sh, bash, tsch)"
The characters ' and \*(L" can be quoted by preceding them with a \e
(backslash); or you can quote a string containing ' with a pair of \*(R"s.
The two alternatives are illustrated by
.Sp
.Vb 2
\&   echo 10d20\e\*(Aq30\e" "20d30\*(Aq40" | GeoConvert \-d \-p \-1
\&   => 10d20\*(Aq30"N 020d30\*(Aq40"E
.Ve
.Sp
Quoting of command line arguments is similar
.Sp
.Vb 2
\&   GeoConvert \-d \-p \-1 \-\-input\-string "10d20\*(Aq30\e" 20d30\*(Aq40"
\&   => 10d20\*(Aq30"N 020d30\*(Aq40"E
.Ve
.IP "Windows command shell (cmd)" 4
.IX Item "Windows command shell (cmd)"
The ' character needs no quoting; the " character can either be quoted
by a ^ or can be represented by typing ' twice.  (This quoting is
usually unnecessary because the trailing designator can be omitted.)
Thus
.Sp
.Vb 2
\&   echo 10d20\*(Aq30\*(Aq\*(Aq 20d30\*(Aq40 | GeoConvert \-d \-p \-1
\&   => 10d20\*(Aq30"N 020d30\*(Aq40"E
.Ve
.Sp
Use \e to quote the " character in a command line argument
.Sp
.Vb 2
\&   GeoConvert \-d \-p \-1 \-\-input\-string "10d20\*(Aq30\e" 20d30\*(Aq40"
\&   => 10d20\*(Aq30"N 020d30\*(Aq40"E
.Ve
.IP "Input from a file" 4
.IX Item "Input from a file"
No quoting need be done if the input from a file.  Thus each line of the
file \f(CW\*(C`input.txt\*(C'\fR should just contain the plain coordinates.
.Sp
.Vb 1
\&  GeoConvert \-d \-p \-1 < input.txt
.Ve
.SH "MGRS"
.IX Header "MGRS"
\&\s-1MGRS\s0 coordinates represent a square patch of the earth, thus \f(CW\*(C`38SMB4488\*(C'\fR
is in zone \f(CW\*(C`38n\*(C'\fR with 444km <= \fIeasting\fR < 445km and 3688km <=
\&\fInorthing\fR < 3689km.  Consistent with this representation,
coordinates are \fItruncated\fR (instead of \fIrounded\fR) to the requested
precision.  Similarly, on input an \s-1MGRS\s0 coordinate represents the
\&\fIcenter\fR of the square (\f(CW\*(C`38n 444500 3688500\*(C'\fR in the example above).
However, if the \fB\-n\fR option is given then the south-west corner of the
square is returned instead (\f(CW\*(C`38n 444000 3688000\*(C'\fR in the example above).
.SH "ZONE"

CHANGELOG.md
## OpenStudio v1.14.0
###### 2016-12-09 — [Diff]( https://github.com/NREL/OpenStudio/compare/v1.13.4...v1.14.0 )

- ![Fixed:][fixed] [#1519]( https://github.com/NREL/OpenStudio/issues/1519 ), People Def can have values for multiple unit types, other loads can't
- ![Fixed:][fixed] [#2178]( https://github.com/NREL/OpenStudio/issues/2178 ), Plenum Space Type - space.spaceType.is_initialized not functioning as expected
- ![Fixed:][fixed] [#2179]( https://github.com/NREL/OpenStudio/issues/2179 ), Plenum Space Type - Odd behavior when rename space type
- ![Fixed:][fixed] [#2299]( https://github.com/NREL/OpenStudio/issues/2299 ), Issue with Zone Sensible Cooling and Heating when heating or cooling only system
- ![Fixed:][fixed] [#2370]( https://github.com/NREL/OpenStudio/issues/2370 ), Error in intersectSurfaces
- ![Fixed:][fixed] [#2372]( https://github.com/NREL/OpenStudio/issues/2372 ), Conversion issues from 1.12.0 to 1.13.0
- ![Fixed:][fixed] [#2376]( https://github.com/NREL/OpenStudio/issues/2376 ), ScheduleRuleset.remove orphans winter and summer profiles

## OpenStudio v1.13.4
###### 2016-12-01 — [Diff]( https://github.com/NREL/OpenStudio/compare/v1.13.3...v1.13.4 )

- ![Fixed:][fixed] [#105]( https://github.com/NREL/OpenStudio/issues/105 ), Deleting Schedule Ruleset object leaves Schedule Day objects behind (Bugzilla #662)
- ![Fixed:][fixed] [#1459]( https://github.com/NREL/OpenStudio/issues/1459 ), LowTemperatureRadiant system input 'Radiant Surface Type' translates unexpected surfaces to SurfaceGroup
- ![Fixed:][fixed] [#1597]( https://github.com/NREL/OpenStudio/issues/1597 ), API allows you to enter invalid object names
- ![Fixed:][fixed] [#2195]( https://github.com/NREL/OpenStudio/issues/2195 ), Timeseries Not Found is being thrown on template reporting measure
- ![Fixed:][fixed] [#2216]( https://github.com/NREL/OpenStudio/issues/2216 ), Spectral Data Set name doesn't forward translate to IDF
- ![Fixed:][fixed] [#2217]( https://github.com/NREL/OpenStudio/issues/2217 ), BCLMeasure should force update if measure.xml missing class_name or other required fields
- ![Fixed:][fixed] [#2241]( https://github.com/NREL/OpenStudio/issues/2241 ), Bad data in energyPlusOutputRequests of reporting measure can break simulation
- ![Fixed:][fixed] [#2251]( https://github.com/NREL/OpenStudio/issues/2251 ), Cannot load openstudio-standards in SketchUp user scripts
- ![Fixed:][fixed] [#2304]( https://github.com/NREL/OpenStudio/issues/2304 ), Missing Objects in EnergyPlus Forward Translator
- ![Fixed:][fixed] [#2312]( https://github.com/NREL/OpenStudio/issues/2312 ), Cannot create custom window construction
- ![Fixed:][fixed] [#2328]( https://github.com/NREL/OpenStudio/issues/2328 ), Need backwards compatible shim for require 'openstudio/ruleset/ShowRunnerOutput'
- ![Fixed:][fixed] [#2331]( https://github.com/NREL/OpenStudio/issues/2331 ), Clean up and embed ruby files in openstudiocore/ruby/openstudio
- ![Fixed:][fixed] [#2332]( https://github.com/NREL/OpenStudio/issues/2332 ), Export zone loads to gbXML
- ![Fixed:][fixed] [#2354]( https://github.com/NREL/OpenStudio/issues/2354 ), ZoneCapacitanceMultiplierResearchSpecial not translated to E+
- ![Fixed:][fixed] [#2362]( https://github.com/NREL/OpenStudio/issues/2362 ), files missing
- ![Fixed:][fixed] [#2368]( https://github.com/NREL/OpenStudio/issues/2368 ), OpenStudio Measure test templates don't work in 2.x
- ![Fixed:][fixed] [#2369]( https://github.com/NREL/OpenStudio/issues/2369 ), OpenStudio 2.x not backwards compatible with 1.x measure tests
- ![Fixed:][fixed] [#2383]( https://github.com/NREL/OpenStudio/issues/2383 ), Refrigeration missing from monthly peak demand
- ![Fixed:][fixed] [#2393]( https://github.com/NREL/OpenStudio/issues/2393 ), Cash flow for fuels not populating because of E+ query change
- ![Fixed:][fixed] [#2394]( https://github.com/NREL/OpenStudio/issues/2394 ), Allow more zone hvac to be drag and dropped from the airloophvac view
- ![Fixed:][fixed] [#2396]( https://github.com/NREL/OpenStudio/issues/2396 ), GeneratorPhotovoltaic::simple constructor makes incomplete object

## OpenStudio v1.13.3
###### 2016-11-10 — [Diff]( https://github.com/NREL/OpenStudio/compare/v1.13.2...v1.13.3 )

- ![Fixed:][fixed] [#2388]( https://github.com/NREL/OpenStudio/pull/2388 ), 131770395 reporting measure 2x support

## OpenStudio v1.13.2
###### 2016-10-29 — [Diff]( https://github.com/NREL/OpenStudio/compare/v1.13.1...v1.13.2 )

- ![Added:][added] [#2382]( https://github.com/NREL/OpenStudio/pull/2382 ), Added support for EMS functionality.
- ![Added:][added] [#2230]( https://github.com/NREL/OpenStudio/pull/2230 ), Added OS:WaterHeater:HeatPump:WrappedCondenser and OS:Coil:WaterHeating:AirToWaterHeatPump:Wrapped objects.
- ![Added:][added] [#2375]( https://github.com/NREL/OpenStudio/pull/2375 ), Added Fuel Type to OS:OtherEquipment:Definition and OS:Coil:Heating:Gas.
- ![Fixed:][fixed] [#2230]( https://github.com/NREL/OpenStudio/pull/2230 ), 2091 wrapped hpwh
- ![Fixed:][fixed] [#2366]( https://github.com/NREL/OpenStudio/pull/2366 ), Set PreferredToolArchitecture = x64 when possible.
- ![Fixed:][fixed] [#2367]( https://github.com/NREL/OpenStudio/pull/2367 ), Updating standard report in pat_app dir
- ![Fixed:][fixed] [#2375]( https://github.com/NREL/OpenStudio/pull/2375 ), Adding Fuel Type to OtherEquipment and Coil:Heating:Gas
- ![Fixed:][fixed] [#2377]( https://github.com/NREL/OpenStudio/issues/2377 ), remove limit on number of fans on air loops
- ![Fixed:][fixed] [#2380]( https://github.com/NREL/OpenStudio/pull/2380 ), removed CAV and VAV fan limit from air loops
- ![Fixed:][fixed] [#2381]( https://github.com/NREL/OpenStudio/pull/2381 ), Pull request for EMS
- ![Fixed:][fixed] [#2382]( https://github.com/NREL/OpenStudio/pull/2382 ), EMS
- ![Fixed:][fixed] [#2386]( https://github.com/NREL/OpenStudio/pull/2386 ), Updates openstudio-standards to 0.1.8 to fix issue hit during PSD tra…

## OpenStudio v1.13.1

openstudiocore/src/geographic_lib/tools/GeoConvert.cpp
/**
 * \file GeoConvert.cpp
 * \brief Command line utility for geographic coordinate conversions
 *
 * Copyright (c) Charles Karney (2008-2015) <charles@karney.com> and licensed
 * under the MIT/X11 License.  For more information, see
 * http://geographiclib.sourceforge.net/
 *
 * See the <a href="GeoConvert.1.html">man page</a> for usage information.
 **********************************************************************/

#include <iostream>
#include <sstream>
#include <string>
#include <sstream>
#include <fstream>
#include <GeographicLib/GeoCoords.hpp>
#include <GeographicLib/DMS.hpp>
#include <GeographicLib/Utility.hpp>
#include <GeographicLib/MGRS.hpp>

#if defined(_MSC_VER)
// Squelch warnings about constant conditional expressions
#  pragma warning (disable: 4127)
#endif

#include "GeoConvert.usage"

int main(int argc, char* argv[]) {
  try {
    using namespace GeographicLib;
    typedef Math::real real;
    Utility::set_digits();
    enum { GEOGRAPHIC, DMS, UTMUPS, MGRS, CONVERGENCE };
    int outputmode = GEOGRAPHIC;
    int prec = 0;
    int zone = UTMUPS::MATCH;
    bool centerp = true, longfirst = false;
    std::string istring, ifile, ofile, cdelim;
    char lsep = ';', dmssep = char(0);
    bool sethemisphere = false, northp = false, abbrev = true;

    for (int m = 1; m < argc; ++m) {
      std::string arg(argv[m]);
      if (arg == "-g")
        outputmode = GEOGRAPHIC;
      else if (arg == "-d") {
        outputmode = DMS;
        dmssep = '\0';
      } else if (arg == "-:") {
        outputmode = DMS;
        dmssep = ':';
      } else if (arg == "-u")
        outputmode = UTMUPS;
      else if (arg == "-m")
        outputmode = MGRS;
      else if (arg == "-c")
        outputmode = CONVERGENCE;
      else if (arg == "-n")
        centerp = false;
      else if (arg == "-z") {
        if (++m == argc) return usage(1, true);
        std::string zonestr(argv[m]);
        try {
          UTMUPS::DecodeZone(zonestr, zone, northp);
          sethemisphere = true;
        }
        catch (const std::exception&) {
          std::istringstream str(zonestr);
          char c;
          if (!(str >> zone) || (str >> c)) {
            std::cerr << "Zone " << zonestr
                      << " is not a number or zone+hemisphere\n";
            return 1;
          }
          if (!(zone >= UTMUPS::MINZONE && zone <= UTMUPS::MAXZONE)) {
            std::cerr << "Zone " << zone << " not in [0, 60]\n";
            return 1;
          }
          sethemisphere = false;
        }
      } else if (arg == "-s") {
        zone = UTMUPS::STANDARD;
        sethemisphere = false;
      } else if (arg == "-t") {
        zone = UTMUPS::UTM;
        sethemisphere = false;
      } else if (arg == "-w")
        longfirst = true;
      else if (arg == "-p") {
        if (++m == argc) return usage(1, true);
        try {
          prec = Utility::num<int>(std::string(argv[m]));
        }
        catch (const std::exception&) {
          std::cerr << "Precision " << argv[m] << " is not a number\n";
          return 1;
        }
      } else if (arg == "-l")
        abbrev = false;
      else if (arg == "-a")
        abbrev = true;
      else if (arg == "--input-string") {
        if (++m == argc) return usage(1, true);
        istring = argv[m];
      } else if (arg == "--input-file") {
        if (++m == argc) return usage(1, true);
        ifile = argv[m];
      } else if (arg == "--output-file") {
        if (++m == argc) return usage(1, true);
        ofile = argv[m];
      } else if (arg == "--line-separator") {
        if (++m == argc) return usage(1, true);
        if (std::string(argv[m]).size() != 1) {
          std::cerr << "Line separator must be a single character\n";
          return 1;
        }
        lsep = argv[m][0];
      } else if (arg == "--comment-delimiter") {
        if (++m == argc) return usage(1, true);
        cdelim = argv[m];
      } else if (arg == "--version") {
        std::cout << argv[0] << ": GeographicLib version "
                  << GEOGRAPHICLIB_VERSION_STRING << "\n";
        MGRS::Check();
        return 0;
      } else
        return usage(!(arg == "-h" || arg == "--help"), arg != "--help");
    }

    if (!ifile.empty() && !istring.empty()) {
      std::cerr << "Cannot specify --input-string and --input-file together\n";
      return 1;
    }
    if (ifile == "-") ifile.clear();
    std::ifstream infile;
    std::istringstream instring;
    if (!ifile.empty()) {
      infile.open(ifile.c_str());
      if (!infile.is_open()) {
        std::cerr << "Cannot open " << ifile << " for reading\n";
        return 1;
      }
    } else if (!istring.empty()) {
      std::string::size_type m = 0;
      while (true) {
        m = istring.find(lsep, m);
        if (m == std::string::npos)
          break;
        istring[m] = '\n';
      }
      instring.str(istring);
    }
    std::istream* input = !ifile.empty() ? &infile :
      (!istring.empty() ? &instring : &std::cin);

    std::ofstream outfile;
    if (ofile == "-") ofile.clear();
    if (!ofile.empty()) {
      outfile.open(ofile.c_str());
      if (!outfile.is_open()) {
        std::cerr << "Cannot open " << ofile << " for writing\n";
        return 1;
      }
    }
    std::ostream* output = !ofile.empty() ? &outfile : &std::cout;

    GeoCoords p;
    std::string s, eol;
    std::string os;
    int retval = 0;

    while (std::getline(*input, s)) {
      eol = "\n";
      try {
        if (!cdelim.empty()) {
          std::string::size_type m = s.find(cdelim);
          if (m != std::string::npos) {
            eol = " " + s.substr(m) + "\n";
            s = s.substr(0, m);
          }
        }
        p.Reset(s, centerp, longfirst);
        p.SetAltZone(zone);
        switch (outputmode) {
        case GEOGRAPHIC:
          os = p.GeoRepresentation(prec, longfirst);
          break;
        case DMS:
          os = p.DMSRepresentation(prec, longfirst, dmssep);
          break;
        case UTMUPS:
          os = (sethemisphere
                ? p.AltUTMUPSRepresentation(northp, prec, abbrev)
                : p.AltUTMUPSRepresentation(prec, abbrev));
          break;
        case MGRS:
          os = p.AltMGRSRepresentation(prec);
          break;
        case CONVERGENCE:
          {
            real
              gamma = p.AltConvergence(),
              k = p.AltScale();
            int prec1 = std::max(-5, std::min(Math::extra_digits() + 8, prec));
            os = Utility::str(gamma, prec1 + 5) + " "
              + Utility::str(k, prec1 + 7);
          }
        }
      }
      catch (const std::exception& e) {
        // Write error message to cout so output lines match input lines
        os = std::string("ERROR: ") + e.what();
        retval = 1;
      }
      *output << os << eol;
    }
    return retval;
  }
  catch (const std::exception& e) {
    std::cerr << "Caught exception: " << e.what() << "\n";
    return 1;
  }
  catch (...) {
    std::cerr << "Caught unknown exception\n";
    return 1;
  }
}
